pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -113,7 +116,7 @@ func (configgen *ConfigGeneratorImpl) BuildDeltaClusters(proxy *model.Proxy, upd
 			svcs, deleted = configgen.deltaFromDestinationRules(key, proxy, subsetClusters)
 		}
 		services = append(services, svcs...)
-		deletedClusters = append(deletedClusters, deleted...)
+		deletedClusters.InsertAll(deleted...)
```

Comment:
I forget why we donot cal removed clusters by  watched resources names minus clusters built